docs/pages/features.typ

```diff
@@ -0,0 +1,470 @@
+#import "../book.typ": *
+
+#show: book-page.with(title: "Formatting Features")
+
+#show: render-examples
+
+= Formatting Features
+
+typstyle follows a consistent set of formatting rules to ensure your Typst code is readable, maintainable, and follows established conventions. This page documents the default formatting styles applied by typstyle.
+
+== Line Width and Indentation
+
+- *Default line width*: 80 characters (configurable with `--line-width`)
+- *Default indentation*: 2 spaces per level (configurable with `--indent-width`)
+- *No tabs*: typstyle always uses spaces for indentation
+
+== Function Calls and Arguments
+
+=== Function Call Arguments
+
+Arguments in function calls are intelligently spaced and aligned. typstyle can handle various argument patterns including trailing commas and complex nesting:
+
+```typst
+#figure(caption: [A very long caption that exceeds the line width],placement:top,supplement:[Figure])
+
+#text(weight:"bold")[Bold text]
+
+#link("http://example.com")[test]
+
+#table(    )[123][456]
+```
+
+=== Flavor Detection
+
+typstyle uses "flavor detection" to determine formatting style. If the first space in arguments contains a newline, arguments are spread across multiple lines:
+
+```typst
+#let my-f(arg1, arg2,
+  args: none) = {
+  arg1 + arg2
+}
+```
+
+=== Comments in Function Calls
+
+Function calls with comments are properly formatted:
+
+```typst
+#{
+  let x = f(
+  cetz.draw.super-long-name.line(
+    start: (0, 0),
+    end: (1, 1),      // note
+  ) // my comment
+)
+}
+```
+
+=== Combinable Expressions
+
+When the only argument is "combinable", it stays on the same line:
```

Comment:
nit: more description about "combinable". Mention "combinable" in document is a good start